### PR TITLE
External Audit Storage: web UI supporting changes

### PR DESCRIPTION
**Pull request by mcbattirola**
PR approved, I'm awaiting for the teleport.e PR to be approved before emerging this one, since this adds a link to a page that will only exist once gravitational/teleport.e#2661 is merged

___

This PR contains changes to enable the  teleport.e PR:https://github.com/gravitational/teleport.e/pull/2661

- New type (ExternalAuditStorage)
- New feature flag
- exporting some components to reuse in the new UI
- refactoring the NoPermBadge into a more generic component (most of the changes are due to this)
- renamed some types (from ExternalCloudAudit to ExternalAuditStorage)

See the teleport.e PR for more context and a demo.

### CODE REVIEW

File: web/packages/teleport/src/services/integrations/types.ts [hunk 1]
```diff
@@ -92,6 +93,24 @@ export function getStatusCodeDescription(
   }
 }
 
+export type ExternalCloudAudit = {
```
Comment (nklaassen) [inline web/packages/teleport/src/services/integrations/types.ts:96]: Can we rename as much as possible to ExternalAuditStorage? I will be going back to rename most of the Go code

Comment (mcbattirola) [reply web/packages/teleport/src/services/integrations/types.ts]: I've renamed all instances in the frontend code and also updated the endpoint paths. Also the context `userAcl` field.

File: web/packages/teleport/src/Integrations/Enroll/IntegrationTiles.tsx [hunk 2]
```diff
@@ -15,53 +15,127 @@
...
+      </IntegrationTile>
+    </>
+  );
+}
+
+function renderExternalAuditStorageBadge(
+  hasExternalAuditStorageAccess: boolean,
+  isEnterprise: boolean
+) {
+  const theme = useTheme();
```
Comment (avatus) [inline web/packages/teleport/src/Integrations/Enroll/IntegrationTiles.tsx:104]: this is called in a non-react function/component

File: web/packages/teleport/src/Integrations/Enroll/IntegrationTiles.tsx [hunk 3]
```diff
@@ -15,53 +15,127 @@
...
-            <div>
-              You don’t have sufficient permissions to create an integration.
-              Reach out to your Teleport administrator to request additional
-              permissions.
-            </div>
-          }
-        />
-      )}
-    </IntegrationTile>
+      color={theme.colors.success}
```
Comment (avatus) [inline web/packages/teleport/src/Integrations/Enroll/IntegrationTiles.tsx:138]:
```suggestion
      color="success"
```

File: web/packages/teleport/src/components/ToolTipBadge/ToolTipBadge.story.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  title: 'Teleport/ToolTip',
+};
+
+export const BadgeString = () => (
+  <SomeBox>
+    I'm a sample container
+    <ToolTipBadge
+      children={'I am a string'}
+      badgeTitle="Title"
+      color="#000000"
```
Comment (avatus) [inline web/packages/teleport/src/components/ToolTipBadge/ToolTipBadge.story.tsx:33]:
```suggestion
      color="success"
```
only if you want it to match the component above

**Review by avatus — APPROVED** [review]
Wasn't able to test locally, just looked at the code. Some small nits

File: web/packages/teleport/src/Integrations/Enroll/IntegrationTiles.tsx (hunk 3 above)
Comment (mcbattirola) [reply web/packages/teleport/src/Integrations/Enroll/IntegrationTiles.tsx]: Thanks!

File: web/packages/teleport/src/Integrations/Enroll/IntegrationTiles.tsx [hunk 5]
```diff
@@ -20,48 +20,119 @@ import { Text, Box } from 'design';
...
+  hasExternalAuditStorage = true,
 }: {
-  hasAccess?: boolean;
+  hasIntegrationAccess?: boolean;
+  hasExternalAuditStorage?: boolean;
 }) {
+  // TODO(mcbattirola): isUsageBasedBilling is used here and in other
+  // parts of the app as synonym with Team product, but this
+  // will change in the future.
+  const isEnterprise = cfg.isEnterprise && !cfg.isUsageBasedBilling;
```
Comment (kimlisa) [inline web/packages/teleport/src/Integrations/Enroll/IntegrationTiles.tsx:43]: are all current usage of `isUsageBasedBilling` mean team right?

**Review by kimlisa — APPROVED** [review]

File: web/packages/teleport/src/Integrations/Enroll/IntegrationTiles.tsx (hunk 5 above)
Comment (mcbattirola) [reply web/packages/teleport/src/Integrations/Enroll/IntegrationTiles.tsx:43]: Yes, right now isUsageBasedBilling === `team` (so the `cfg.isEnterprise` is redundant, but I wanted to be explicit).

**Pull request merged**